ocpp-v2/src/main/java/eu/chargetime/ocpp/v201/feature/function/ClientProvisioningFunction.java

```diff
@@ -0,0 +1,115 @@
+/*
+   ChargeTime.eu - Java-OCA-OCPP
+
+   MIT License
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE.
+*/
+
+package eu.chargetime.ocpp.v201.feature.function;
+
+import eu.chargetime.ocpp.feature.FunctionFeature;
+import eu.chargetime.ocpp.feature.function.Function;
+import eu.chargetime.ocpp.model.Confirmation;
+import eu.chargetime.ocpp.model.Request;
+import eu.chargetime.ocpp.v201.feature.*;
+import eu.chargetime.ocpp.v201.model.messages.*;
+import eu.chargetime.ocpp.v201.model.types.*;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.UUID;
+
+/** Class with client request creators and handlers for the Provisioning functional block. */
+public class ClientProvisioningFunction implements Function {
+
+  private final ClientProvisioningEventHandler eventHandler;
+  private final ArrayList<FunctionFeature> features;
+
+  public ClientProvisioningFunction(ClientProvisioningEventHandler eventHandler) {
+    this.eventHandler = eventHandler;
+    features = new ArrayList<>();
+    features.add(new BootNotificationFeature(null));
+    features.add(new GetBaseReportFeature(this));
+    features.add(new GetReportFeature(this));
+    features.add(new GetVariablesFeature(this));
+    features.add(new HeartbeatFeature(null));
+    features.add(new NotifyReportFeature(null));
```

Comment:
Chapter B. Provisioning. NotifyReport is needed to respond to GetBaseReport requests. Heartbeat is actually described in Chapter G. in the specification, but since the BootNotification response has a parameter that sets the Heartbeat interval, I put it in here, so that when you implement BootNotification, Heartbeat comes with it.